src/ImageCanvas/region-tools.js

```diff
@@ -47,10 +47,21 @@ export type Polygon = {|
   open?: boolean,
   points: Array<[number, number]>
 |}
+export type Circle = {|
+  ...$Exact<BaseRegion>,
+  type: "circle",
+  // radius: number,
+  // x and y indicate the coordinates of the centre of the circle
+  x: number,
+  y: number,
+  // xr and yr indicate the x and y coordinates of the current mouse pointer while dragging.
+  xr: number,
+  yr: number
+|}
 
 export type Region = Point | PixelRegion | Box | Polygon
 
-export const getEnclosingBox = (region: Region) => {
+export const getEnclosingBox = (region: Region, iw: number, ih: number) => {
```

Comment:
it's unclear to me why `iw` and `ih` are needed (because they're not needed for other structures). It seems like the enclosing box should be equal to...

```
{
  x: region.x - region.xr,
  y: region.y - region.yr,
  w: region.xr,
  h: region.yr
}
```

If possible, we should remove `iw` and `ih` from the parameters here.

Reply:
As I mentioned in my previous comment, xr and yr aren't actually representing the radius exactly but are representing the coordinates to last point while dragging by a mouse. So, I need ih and iw.
`
{
  x: region.x - region.xr,
  y: region.y - region.yr,
  w: region.xr,
  h: region.yr
}
`

This works if xr is x-radius and yr is y-radius. 
I'm thinking to make xr as x-radius and yr as y-radius but then, I'd need ih and iw in the reducer. I'm new to react and redux. So, I'd need your advice here as to how I can set ih and iw in the state (store) of the reducer.

**Reason I need ih and iw in the reducer:**
Based on the center of the circle and the current point, I have to calculate x-radius and y-radius.